packages/dd-trace/src/plugins/log_plugin.js

```diff
@@ -43,12 +43,21 @@ module.exports = class LogPlugin extends Plugin {
       this.tracer.inject(span, LOG, holder)
       arg.message = messageProxy(arg.message, holder)
     })
+    this.structured = false // default unstructured logger
+  }
+
+  _injectLogs (config) {
+    if (this.structured) {
+      return (config.logInjection === true || config.logInjection === 'structured')
+    }
+
+    return config.logInjection === true
```

Comment:
The whole function body can be replaced by `config.logInjection === true || (this.structured && config.logInjection === 'structured')`